graf2d/gpadv7/inc/ROOT/RPadBase.hxx

```diff
@@ -81,43 +83,45 @@ public:
    /// \returns vector of vector (ret[x][y]) of created pads.
    std::vector<std::vector<std::shared_ptr<RPad>>> Divide(int nHoriz, int nVert, const RPadExtent &padding = {});
 
-   /// Create drawable of specified class T
-   template<class T, class... ARGS>
-   auto Draw(ARGS... args)
+   /// Add object to be painted.
+   /// Correspondent drawable will be created via GetDrawable() function which should be defined and be accessed at calling time.
+   /// If required, extra arguments for GetDrawable() function can be provided.
+   template <class T, class... ARGS>
+   auto Draw(const std::shared_ptr<T> &what, ARGS... args)
    {
-      auto drawable = std::make_shared<T>(args...);
+      // Requires GetDrawable(what) to be known!
+      auto drawable = GetDrawable(what, args...);
 
       TestIfFrameRequired(drawable.get());
 
-      fPrimitives.emplace_back(drawable);
+      AddPrimitive(drawable);
 
       return drawable;
    }
 
-   /// Add existing drawable instance to canvas
-   auto Draw(std::shared_ptr<RDrawable> &&drawable)
+   /// Create drawable of specified class T
+   template<class T, class... ARGS>
+   std::shared_ptr<T> Draw(ARGS... args)
```

Comment:
```suggestion
   template<class T, class... ARGS, class = std::enable_if<std::is_base_of<RDrawable, std::remove_cv<T>::type>::type>
   std::shared_ptr<T> Draw(ARGS... args)
```

Reply:
This does not work. You have typo somewhere.

Reply:
Indeed, fixed.

Reply:
This change mix-up PyROOT - it fails to find proper signature. I will revert change

Reply:
OK so that's broken in ROOT's meta belly.

What about
```c++
// SFINAE-enabled only if T derives from RDrawable; returns std::shared_ptr<T>
template<class T, class... ARGS>
typename std::enable_if<std::is_base_of<RDrawable, typename std::remove_cv<T>::type>::value, std::shared_ptr<T>>::type
Draw(ARGS... args)
```
Once we have C++20... we can use `requires` clauses.